pkg/operator2/route.go

```diff
@@ -18,13 +18,43 @@ func (c *authOperator) handleRoute() (*routev1.Route, error) {
 	if err != nil {
 		return nil, err
 	}
-	if len(route.Spec.Host) == 0 {
-		return nil, fmt.Errorf("route has no host: %#v", route)
+
+	if err := isValidRoute(route); err != nil {
+		return nil, err
 	}
-	// TODO make sure route is correct
+
 	return route, nil
 }
 
+func isValidRoute(route *routev1.Route) error {
```

Comment:
It might be simpler to do some form of deep equal.

Reply:
Did not find a deep-equal that would only check certain fileds. We might possibly add a for-loop which would use `reflect.ValueOf()` comparison, although this here seems a bit more explicit.

Nevertheless, I modified the check a bit so that it does not use hard-coded values and uses whatever is in `defaultRoute()` instead.